domain/model/state/controllerstate.go

```diff
@@ -972,21 +983,82 @@ func (s *State) ListModelIDs(ctx context.Context) ([]coremodel.UUID, error) {
 		return nil, errors.Capture(err)
 	}
 
-	var models []coremodel.UUID
+	var dbResult []dbUUID
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		var result []dbUUID
-		if err := tx.Query(ctx, stmt).GetAll(&result); errors.Is(err, sqlair.ErrNoRows) {
-			return nil
-		} else if err != nil {
+		err := tx.Query(ctx, stmt).GetAll(&dbResult)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
 			return errors.Errorf("getting all model UUIDs: %w", err)
 		}
+		return nil
+	})
 
-		for _, r := range result {
-			models = append(models, coremodel.UUID(r.UUID))
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	modelUUIDs := make([]coremodel.UUID, 0, len(dbResult))
+	for _, r := range dbResult {
+		modelUUIDs = append(modelUUIDs, coremodel.UUID(r.UUID))
+	}
+	return modelUUIDs, nil
+}
+
+// ListModelUUIDsForUser returns a list of all the model uuids that a user has
+// access to in the controller.
+// The following errors can be expected:
+// - [accesserrors.UserNotFound] when the user does not exist.
+func (s *State) ListModelUUIDsForUser(
+	ctx context.Context,
+	userUUID user.UUID,
+) ([]coremodel.UUID, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	userUUIDVal := dbUUID{UUID: userUUID.String()}
+	stmt, err := s.Prepare(`
+SELECT &dbUUID.*
+FROM v_model
+WHERE owner_uuid = $dbUUID.uuid
```

Comment:
Leave out the owner bit. The user who created the model will be in the access list.
And owner is going away.

Reply:
Done & Done